app/routes/devices.$id.hardware.$hardware_name.history.tsx

```diff
@@ -82,6 +93,11 @@ export default function Hardware() {
     hardwareHistory[0].data,
   ).filter((property) => excludedProperties.includes(property) === false);
 
+  // shape of data is not ideal here, we have a struct mapping event name to
+  // array of events. Here we drop name, probably change to backend could keep
+  // the name in the struct
+  const allEvents = flatMap(events);
+
```

Comment:
Interesting, did you test this change with a chart that shows multiple vials' histories?